### PR TITLE
Fix fp-ts deprecation warnings in equality examples

### PR DESCRIPTION
**Pull request by adomokos**

### CODE REVIEW

File: docs/equality.md [hunk 1]
```diff
@@ -7,28 +7,28 @@ We show the most common usages here, but if you need more ways to check for equa
...
 type Point = {
   x: number;
   y: number;
 };
 
-const eqPoint: eq.Eq<Point> = eq.getStructEq({
-  x: eq.eqNumber,
-  y: eq.eqNumber,
-});
+const eqPoint: Eq<Point> = {
```
Comment (grossbart) [inline docs/equality.md]: We can use https://gcanti.github.io/fp-ts/modules/Eq.ts.html#struct here.

File: docs/equality.md [hunk 2]
```diff
@@ -7,28 +7,28 @@ We show the most common usages here, but if you need more ways to check for equa
...
+date.Eq.equals(new Date("1984-01-27"), new Date("1984-01-27")); // true
+number.Eq.equals(3, 3); // true
+string.Eq.equals("Cyndi", "Cyndi"); // true
 ```
 
 ## Compare structures
 
 ```ts
 import { eq } from "fp-ts";
+import { Eq } from "fp-ts/lib/Eq";
```
Comment (grossbart) [inline docs/equality.md]: We should not import from `lib` anymore, the following should work:

```suggestion
import { Eq } from "fp-ts/Eq";
```

File: docs/equality.md [hunk 3]
```diff
@@ -37,26 +37,26 @@ This structure can be combined further:
...
```
Comment (grossbart) [inline docs/equality.md]:
```suggestion
import { Eq } from "fp-ts/Eq";
```

File: docs/equality.md [hunk 4]
```diff
@@ -37,26 +37,26 @@ This structure can be combined further:
...
 type Point = {
   x: number;
   y: number;
 };
 
-const eqPoint: eq.Eq<Point> = eq.getStructEq({
-  x: eq.eqNumber,
-  y: eq.eqNumber,
-});
+const eqPoint: Eq<Point> = {
```
Comment (grossbart) [inline docs/equality.md]: We can use https://gcanti.github.io/fp-ts/modules/Eq.ts.html#struct here.

File: docs/equality.md [hunk 5]
```diff
@@ -37,26 +37,26 @@ This structure can be combined further:
...
 type Vector = {
   from: Point;
   to: Point;
 };
 
-const eqVector: eq.Eq<Vector> = eq.getStructEq({
-  from: eqPoint,
-  to: eqPoint,
-});
+const eqVector: Eq<Vector> = {
```
Comment (grossbart) [inline docs/equality.md]: We can use https://gcanti.github.io/fp-ts/modules/Eq.ts.html#struct here.

File: docs/equality.md [hunk 6]
```diff
@@ -67,9 +67,9 @@ eqVector.equals(
 ## Compare arrays
 
 ```ts
-import { array, eq } from "fp-ts";
+import { array, string } from "fp-ts";
 
-const eqArrayOfStrings = array.getEq(eq.eqString);
+const eqArrayOfStrings = array.getEq<string>(string.Eq);
```
Comment (grossbart) [inline docs/equality.md]: I'm surprised that we need to provide the generic value here, doesn't this work without `<string>`? (I haven't tried it)

File: docs/equality.md [hunk 7]
```diff
@@ -78,18 +78,18 @@ Test the equality of structures nested within arrays:
 
 ```ts
 import { array, eq } from "fp-ts";
+import { Eq } from "fp-ts/lib/Eq";
```
Comment (grossbart) [inline docs/equality.md]:
```suggestion
import { Eq } from "fp-ts/Eq";
```

File: docs/equality.md [hunk 8]
```diff
@@ -78,18 +78,18 @@ Test the equality of structures nested within arrays:
...
 type Point = {
   x: number;
   y: number;
 };
 
-const eqPoint: eq.Eq<Point> = eq.getStructEq({
-  x: eq.eqNumber,
-  y: eq.eqNumber,
-});
+const eqPoint: Eq<Point> = {
```
Comment (grossbart) [inline docs/equality.md]: We can use https://gcanti.github.io/fp-ts/modules/Eq.ts.html#struct here.

File: docs/equality.md [hunk 9]
```diff
@@ -78,18 +78,18 @@ Test the equality of structures nested within arrays:
...
+  equals: (p1: Point, p2: Point) => p1 === p2 || (p1.x === p2.x && p1.y === p2.y)
+}
 
-const eqArrayOfPoints: eq.Eq<Array<Point>> = array.getEq(eqPoint);
+const eqArrayOfPoints = array.getEq<Point>(eqPoint);
```
Comment (grossbart) [inline docs/equality.md]: Does this also work without providing the generic variable? Same as further up, I would hope that it is not necessary anymore.

**Review by grossbart — CHANGES_REQUESTED** [review]
Thanks for updating this! I have added some small things that should be fixed before merging. If you find the time to do these, that would be great, if not I will try to fix it later.

File: docs/equality.md (hunk 9 above)
Comment (adomokos) [reply docs/equality.md]: I'll check.

File: docs/equality.md (hunk 1 above)
Comment (adomokos) [reply docs/equality.md]: Fixed it.

File: docs/equality.md (hunk 5 above)
Comment (adomokos) [reply docs/equality.md]: Fixed it.

File: docs/equality.md (hunk 6 above)
Comment (adomokos) [reply docs/equality.md]: Removed.

File: docs/equality.md (hunk 8 above)
Comment (adomokos) [reply docs/equality.md]: Fixed it.

File: docs/equality.md (hunk 9 above)
Comment (adomokos) [reply docs/equality.md]: Removed.

**Comment by adomokos** [comment]
@grossbart, thanks for the improvement suggestions. Please take a look again, hope this is polished enough to merge it.

**Review by grossbart — APPROVED** [review]
Thanks for the quick polishing and your initiative, will merge right now :)

**Pull request merged**